client/src/main/java/org/apache/falcon/entity/v0/EntityType.java

```diff
@@ -124,4 +128,27 @@ public static EntityType getEnum(String type) {
                     + Arrays.toString(values()).toLowerCase() + ".");
         }
     }
+
+    /**
+     * Finds the value of the given enumeration by name, case-insensitive.
+     * Throws an IllegalArgumentException if no match is found.
+     *
+     * @param entityType : string to compare against the value of EntityType
+     * @return EntityType : Object of type EntityType
+     * @throws IllegalArgumentException incase the argument is not present in the EntityType enum
+     **/
+
+    public static EntityType fromString(String entityType) {
```

Comment:
getEnum will work perfectly here. There is no need to make the changes we are writing to DB ,the purpose of this method was to introduce caseInsensitivity while getting the EnumType it can easily be served by getEnum.